src/seedu/addressbook/data/person/Address.java

```diff
@@ -8,33 +8,37 @@
  */
 public class Address {
 
-    public static final String EXAMPLE = "123, some street";
-    public static final String MESSAGE_ADDRESS_CONSTRAINTS = "Person addresses can be in any format";
-    public static final String ADDRESS_VALIDATION_REGEX = ".+"; // TODO test regex  ^a\/([^,]+,){1-3}\d*
+    public static final String EXAMPLE = "[block], street, [unit], postal_code";
+    public static final String MESSAGE_ADDRESS_CONSTRAINTS = "Person addresses must be in the format '" + EXAMPLE +"'";
+    public static final String ADDRESS_VALIDATION_REGEX = "^[^,]*(, [^,]*){3}$";
 
     private boolean isPrivate;
     private Block block;
     private Street street;
     private Unit unit;
     private PostalCode postalCode;
 
+    private static class AddressCSVIndices {
```

Comment:
this name is a bit iffy.
i get where you are coming from, but seems like you are introducing something complex.
try to think of a better name, if possible?